### PR TITLE
Enhance UI

### LINKED ISSUE #129: Integrating time-series-buffer and AgentBuffer
Since the implementation of AgentBuffer as part of every agent now, we should think of a way for enabling metrological buffering and suggest that:
- Extend a class MetrologicalAgentBuffer which adds constraints on the data with uncertainties attached and metadata
- utilise time-series-buffer as part of it
- rework tutorial 4 to be consistent with these changes above

### PR DESCRIPTION
**Pull request by bangxiangyong**
Multiple updates to the front-end UI display. e.g inclusion of agent parameters on side panel

Updates from https://github.com/Met4FoF/agentMET4FOF/pull/188 are already merged into this branch, so this should be considered instead of it, thus this resolves #129.

### CODE REVIEW

**Review by BjoernLudwigPTB — COMMENTED** [review]
When I call _basic_send_plot_ I can see shortly after starting the execution:

```shell
Exception ignored in: <module 'threading' from '/usr/lib/python3.8/threading.py'>
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 1388, in _shutdown
    lock.acquire()
KeyboardInterrupt: 
NS shut down.
```

but then at least three plots appear and the network runs with updating plots. Is this expected? I love what I remember, that you showed to us, but unfortunately I am not able to run it. I still had some remarks on some parts of the implementation. Hope you find this useful. As soon, as I am able to run the "new" dashboard, I would love to take a closer look on the actual features.

File: develop_UI/agent_module_example.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from agentMET4FOF.agents import AgentMET4FOF
```
[the remark this answers is not in the document]
Comment (bangxiangyong) [reply develop_UI/agent_module_example.py]: yes, you are right..  i just refactored it into the tutorial folder.

File: agentMET4FOF/dashboard/Dashboard_agt_net.py [hunk 2]
```diff
@@ -599,6 +658,18 @@ def plot_monitor_graphs(n_interval):
             # set dimensions of each monitor agent's graph
             return [all_graphs]
 
+        @app.callback(
+            dash.dependencies.Output('toast-js-script', 'run'),
+            [dash.dependencies.Input('interval-update-toast', 'n_intervals')])
+        def myfun(n_intervals):
```
[the remark this answers is not in the document]
Comment (bangxiangyong) [reply agentMET4FOF/dashboard/Dashboard_agt_net.py]: good catch.. i renamed it

File: agentMET4FOF/dashboard/LayoutHelper.py [hunk 3]
```diff
@@ -24,10 +25,13 @@ def create_nodes_cytoscape(agent_graph):
 
     return new_elements
 
-def create_edges_cytoscape(edges):
+def create_edges_cytoscape(edges, hide_default_edge=True):
```
[the remark this answers is not in the document]
Comment (bangxiangyong) [reply agentMET4FOF/dashboard/LayoutHelper.py:28]: done in the latest commit

File: develop_UI/agent_main_example.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    sine_agent = agent_network.add_agent(agentType=SineGeneratorAgent)
+
+    coalition = agent_network.add_coalition(agents=[])
+
+    agent_network.add_coalition_agent(name=coalition.name,agents=[sine_agent])
+
+    return agent_network
+
+if __name__ == "__main__":
+    demonstrate_generator_agent_use()
```
[the remark this answers is not in the document]
Comment (bangxiangyong) [reply develop_UI/agent_main_example.py]: yes it's meant as a minimum example showcasing added ParameterisedSineGneartorAgent at the right panel.
and i fixed the bug so it should work now..

File: agentMET4FOF/dashboard/Dashboard_agt_net.py [hunk 5]
```diff
@@ -553,16 +612,16 @@ def plot_monitor_graphs(n_interval):
             if agentNetwork.get_mode() != "Running" and n_interval > 0:
                 raise PreventUpdate
 
-            agent_names = agentNetwork.agents()  # get all agent names
-            agent_type = "Monitor"  # all agents with Monitor in its name will be selected
+            agent_type = "Monitor"
```
[the remark this answers is not in the document]
Comment (bangxiangyong) [reply agentMET4FOF/dashboard/Dashboard_agt_net.py]: yup, done.

**Comment by bangxiangyong** [comment]
Regarding the `basic_send_plot` error, could be a bug on your end, it's working fine on my PC with the 3 plots and no error msg..

**Pull request merged**